servicetalk-http-router-jaxrs/src/main/java/io/servicetalk/http/router/jaxrs/HttpJaxRsRouterBuilder.java

```diff
@@ -0,0 +1,255 @@
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.BiPredicate;
+import javax.annotation.Nonnull;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.CookieParam;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.FormParam;
+import javax.ws.rs.HeaderParam;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Application;
+import javax.ws.rs.core.Context;
+
+import static javax.ws.rs.core.MediaType.WILDCARD;
+
+/**
+ * Builds an {@link StreamingHttpService} which routes requests to JAX-RS annotated classes,
+ * using internal routing engine, eg.
+ * <pre>{@code
+ * final StreamingHttpService router = new HttpJaxRsRouterBuilder()
+ *     .buildStreaming(application);
+ * }</pre>
+ */
+public final class HttpJaxRsRouterBuilder {
+
+    /**
+     * Build the {@link HttpService} for the specified JAX-RS {@link Application}.
+     *
+     * @param application the {@link Application} to route requests to.
+     * @return the {@link HttpService}.
+     */
+    public HttpService build(final Application application) {
+        return toAggregated(from(application));
+    }
+
+    /**
+     * Build the {@link StreamingHttpService} for the specified JAX-RS {@link Application}.
+     *
+     * @param application the {@link Application} to route requests to.
+     * @return the {@link StreamingHttpService}.
+     */
+    public StreamingHttpService buildStreaming(final Application application) {
+        return from(application);
+    }
+
+    /**
+     * Build the {@link BlockingHttpService} for the specified JAX-RS {@link Application}.
+     *
+     * @param application the {@link Application} to route requests to.
+     * @return the {@link BlockingHttpService}.
+     */
+    public BlockingHttpService buildBlocking(final Application application) {
+        return toBlocking(from(application));
+    }
+
+    /**
+     * Build the {@link BlockingStreamingHttpService} for the specified JAX-RS {@link Application}.
+     *
+     * @param application the {@link Application} to route requests to.
+     * @return the {@link BlockingStreamingHttpService}.
+     */
+    public BlockingStreamingHttpService buildBlockingStreaming(final Application application) {
+        return toBlockingStreaming(from(application));
+    }
+
+    public StreamingHttpService from(final Application application) {
+
+        HttpPredicateRouterBuilder routerBuilder = new HttpPredicateRouterBuilder();
+
+        if (!application.getClasses().isEmpty()) {
+            throw new UnsupportedOperationException("Method application.getClasses() is not supported. " +
+                    "Use application.getSingletons() instead");
+        }
+
+        for (Object resource : application.getSingletons()) {
+
+            final Path rootPath = resource.getClass().getAnnotation(Path.class);
```

Comment:
I think that the JAX-RS annotation inheritance rules should be supported to follow the principle of least surprise for users.